uniswap-v2/src/modules/4_map_pool_events.rs

```diff
@@ -0,0 +1,175 @@
+use substreams::scalar::BigInt;
+use substreams::store::StoreGetProto;
+use substreams::store::{DeltaBigInt, Deltas, StoreGet};
+use substreams_ethereum::pb::eth::v2::{self as eth};
+use substreams_helper::event_handler::EventHandler;
+use substreams_helper::hex::Hexable;
+
+use crate::abi::Pool::events::{Burn, Mint, Swap, Sync};
+use crate::common::traits::PoolAddresser;
+use crate::pb::uniswap::v2::event::Type::{DepositType, SwapType, SyncType, WithdrawType};
+use crate::pb::uniswap::v2::{DepositEvent, SwapEvent, SyncEvent, WithdrawEvent};
+use crate::pb::uniswap::v2::{Event, Events, Pool};
+use crate::store_key::StoreKey;
+use crate::utils;
+
+#[substreams::handlers::map]
+pub fn map_pool_events(
+    block: eth::Block,
+    pools_store: StoreGetProto<Pool>,
+    balance_deltas: Deltas<DeltaBigInt>,
+) -> Result<Events, substreams::errors::Error> {
+    let mut events = vec![];
+
+    handle_mint(&block, &pools_store, &balance_deltas, &mut events);
```

Comment:
Check here: https://github.com/messari/substreams/blob/master/substreams-helper/src/event_handler.rs#L13-L19

Can you change these fns to just set up the handlers instead for the different events and then call `eh.handle_events();` after. This way you will loop through all the logs just once

Reply:
As I have to mutably store all the events in a list, I have to do it in separate functions. 

Ref: https://github.com/messari/substreams/blob/master/substreams-helper/src/event_handler.rs#L22-L39